examples/src/big_compute_buffers/mod.rs

```diff
@@ -0,0 +1,288 @@
+    let storage_buffers = create_storage_buffers(device, numbers, input_size);
+
+    let (bind_group_layout, bind_group) = setup_binds(&storage_buffers, device);
+
+    let compute_pipeline = setup_pipeline(device, bind_group_layout, cs_module);
+    (
+        staging_buffers,
+        storage_buffers,
+        bind_group,
+        compute_pipeline,
+    )
+}
+
+fn setup_pipeline(
+    device: &wgpu::Device,
+    bind_group_layout: wgpu::BindGroupLayout,
+    cs_module: wgpu::ShaderModule,
+) -> wgpu::ComputePipeline {
+    let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+        label: Some("Compute Pipeline Layout"),
+        bind_group_layouts: &[&bind_group_layout],
+        push_constant_ranges: &[],
+    });
+
+    device.create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+        label: Some("Compute Pipeline"),
+        layout: Some(&pipeline_layout),
+        module: &cs_module,
+        entry_point: Some("main"),
+        compilation_options: Default::default(),
+        cache: None,
+    })
+}
+
+fn setup_binds(
+    storage_buffers: &[wgpu::Buffer],
+    device: &wgpu::Device,
+) -> (wgpu::BindGroupLayout, wgpu::BindGroup) {
+    let bind_group_entries: Vec<wgpu::BindGroupEntry> = storage_buffers
+        .iter()
+        .enumerate()
+        .map(|(bind_idx, buffer)| wgpu::BindGroupEntry {
+            binding: bind_idx as u32,
+            resource: buffer.as_entire_binding(),
+        })
+        .collect();
+
+    let bind_group_layout_entries: Vec<wgpu::BindGroupLayoutEntry> = (0..storage_buffers.len())
+        .map(|bind_idx| wgpu::BindGroupLayoutEntry {
+            binding: bind_idx as u32,
+            visibility: wgpu::ShaderStages::COMPUTE,
+            ty: wgpu::BindingType::Buffer {
+                ty: wgpu::BufferBindingType::Storage { read_only: false },
+                has_dynamic_offset: false,
+                min_binding_size: None,
+            },
+            count: Some(NonZeroU32::new(1).unwrap()),
+        })
+        .collect();
+
+    let bind_group_layout = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+        label: Some("Custom Storage Bind Group Layout"),
+        entries: &bind_group_layout_entries,
+    });
+
+    let bind_group = device.create_bind_group(&wgpu::BindGroupDescriptor {
+        label: Some("Combined Storage Bind Group"),
+        layout: &bind_group_layout,
+        entries: &bind_group_entries,
+    });
+    (bind_group_layout, bind_group)
+}
+
+fn calculate_chunks(numbers: &[f32], max_buffer_size: u64) -> Vec<&[f32]> {
+    let max_elements_per_chunk = max_buffer_size as usize / std::mem::size_of::<f32>(); // Calculate max f32 elements per buffer
+    numbers.chunks(max_elements_per_chunk).collect()
+}
+
+fn create_storage_buffers(
+    device: &wgpu::Device,
+    numbers: &[f32],
+    input_size: u64, // bytes..
+) -> Vec<wgpu::Buffer> {
+    if input_size > MAX_BUFFER_SIZE {
```

Comment:
Do we really need to have separate paths for small and large inputs? Won't the path that calls `calculate_chunks` work fine in both cases?